api/v1alpha8/openstackcluster_webhook.go

```diff
@@ -63,6 +63,14 @@ func (r *OpenStackCluster) ValidateCreate() (admission.Warnings, error) {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "identityRef", "kind"), "must be a Secret"))
 	}
 
+	if r.Spec.ManagedSecurityGroups.Enabled && r.Spec.ManagedSecurityGroups.AllNodesSecurityGroupRules != nil {
+		for _, rule := range r.Spec.ManagedSecurityGroups.AllNodesSecurityGroupRules {
+			if rule.RemoteManagedGroups == nil {
```

Comment:
* don't check nil, but use length
* remove that check
* check mutually exclusive fields: remoteIPPrefix, remoteSecurityGroupID and remoteManagedGroups can't be set together.